src/v/kafka/requests/response_writer.h

```diff
@@ -184,14 +184,8 @@ class response_writer {
     // wrap a writer in a kafka bytes array object. the writer should return
     // true if writing no bytes should result in the encoding as nullable bytes,
     // and false otherwise.
-    //
-    // clang-format off
     template<typename ElementWriter>
-    CONCEPT(requires requires (ElementWriter writer,
-                               response_writer& rw) {
-        { writer(rw) } -> bool;
-    })
-    // clang-format on
+    CONCEPT(requires std::predicate<ElementWriter, response_writer&>)
```

Comment:
This doesn't seem right, `std::predicate` refines `std::regular_invocable`, which "requires the invocation to not modify either the callable object or the arguments and be equality-preserving".

Reply:
what do you propose? that it takes a const reference as a parameter?

Reply:
Conceptually it feels like it modifies the writer. Probably roughly how it was is fine for now.

Reply:
ahh i see. you're right. that isn't a std::predicate because of the non-constness. i changed it back.